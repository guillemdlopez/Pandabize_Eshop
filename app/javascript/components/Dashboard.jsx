import React, { useContext, useMemo, useState } from "react";
import OrdersList from "./OrdersList";
import Sidebar from "./Sidebar";

import { API_USERS_URL } from "../fixtures/variables";
import { userContext } from "../context/userContext";

const Dashboard = () => {
  const [userInfo, setUserInfo] = useState({});

  const { userId } = useContext(userContext);

  const getUserInfo = async () => {
    try {
      const getJSON = await fetch(`${API_USERS_URL}${userId}`);
      if (!getJSON.ok) throw new Error(`Problem getting location data`);

      const res = await getJSON;
      if (!res.ok) throw new Error(`Bicycle not found ${res.status}`);

      const data = await res.json();
      setUserInfo(data);
    } catch (err) {
      console.error(err);
    }
  };
  useMemo(() => getUserInfo(), [userId]);
  const currentUser = userInfo?.user;

  return (
    <div className="container animate__animated animate__fadeIn">
      <div className="row mt-5">
        <div className="col-lg-4 col-md-12 mb-4">
          <Sidebar {...currentUser} orders={userInfo.orders} />
        </div>
        <div className="col-lg-8 col-md-12">
          <OrdersList orders={userInfo.orders} />
        </div>
      </div>
    </div>
  );
};

export default Dashboard;
