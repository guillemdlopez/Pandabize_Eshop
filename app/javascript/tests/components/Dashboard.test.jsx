import { mount } from "enzyme";
import React from "react";
import Dashboard from "../../components/Dashboard";
import { userContext } from "../../context/userContext";

describe("Testing the <Dashboard/> component", () => {
  test("should render the component correctly", () => {
    const wrapper = mount(
      <userContext.Provider value={{ user: "Guillem", userId: 10 }}>
        <Dashboard />
      </userContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
