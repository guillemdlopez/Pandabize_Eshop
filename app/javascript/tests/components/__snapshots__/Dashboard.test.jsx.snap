// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the <Dashboard/> component should render the component correctly 1`] = `
<div
  className="container animate__animated animate__fadeIn"
>
  <div
    className="row mt-5"
  >
    <div
      className="col-lg-5 col-md-12 mb-4"
    >
      <div
        className="sidebar d-flex align-items-center"
      >
        <img
          className="avatar-md"
        />
        <div
          className="user-info"
        >
          <ul
            className="sidebar-list ml-3"
          >
            <li
              className="full_name-item"
            >
              <strong>
                Full Name: 
              </strong>
              undefined undefined
            </li>
            <li
              className="orders-item"
            >
              <strong>
                Orders:
              </strong>
               
            </li>
            <li>
              <button
                className="btn btn-outline-success btn-sm"
                onClick={[Function]}
              >
                Continue shopping
              </button>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      className="col-lg-7 col-md-12"
    />
  </div>
</div>
`;
